python/ray/tests/test_runtime_env.py

```diff
@@ -772,6 +772,38 @@ def test_container_option_serialize():
     assert job_config_serialized.count(b"image") == 2
 
 
+def test_working_dir_override_failure(shutdown_only):
+    ray.init()
+
+    @ray.remote(runtime_env={"working_dir": "."})
+    def f():
+        pass
+
+    with pytest.raises(NotImplementedError):
+        f.remote()
```

Comment:
any reason this doesn't happen at definition time?

Reply:
because the args in the `@ray.remote` decorators are applied lazily and only injected at submission time :-) so here have the same code path for both decorator path and options path